### PR TITLE
[LTC] Pass a BackendDevice parameter into GetIrValueForScalarFromCodegen

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82970

Summary: Currently GetIrValueForScalarFromCodegen uses CPU as the
default backend device for scalars, but we should make it a
backend-dependent decision.

### CODE REVIEW

**Comment by desertfire** [comment]
For https://github.com/pytorch/pytorch/issues/82208. @JackCaoG , sorry for the delay because I was on PTO last week. Here is a fix on the TS side. Let me know if it can unblock the XLA side.

**Review by Krovatkin — APPROVED** [review]
:shipit:

I think this is a reasonable fix as long as it works for @JackCaoG w.r.t. possible overheads.

**Comment by JackCaoG** [comment]
Thanks Bin, I will give this a try today!

**Review by JackCaoG — APPROVED** [review]
Thanks @desertfire. My local test passed, waiting for CI for https://github.com/pytorch/xla/pull/3768 to confirm.

**Pull request closed without merging**
